Separate out NumberFormatToString

zbraniecki opened the pull request:
Following https://github.com/tc39/ecma402/commit/2f9226d2d913c12e62cf0e9909401fcba638cae2 this change prepares Intl.js for PluralRules.

caridy commented:
@zbraniecki I will really like to keep the algo steps comments for easy evolution of those methods, you can do that either manually (if it is a small algo) or using the tool.

zbraniecki commented:
@caridy - I used the tool and also cleaned up some comments in PartitionNumberFormat which were not incrementing and fixed alignment in the function which was broken.

caridy commented:
awesome, thanks! that's precisely why I want to use this :)
